app/src/pages/inside/stepPage/modals/makeDecisionModal/executionSection/executionSection.jsx

```diff
@@ -0,0 +1,204 @@
+        );
+        setModalState({
+          currentTestItems: currentItemsWithLogs,
+        });
+        setCurrentItemsLoading(false);
+      })
+      .catch(({ message }) => {
+        setCurrentItemsLoading(false);
+        dispatch(
+          showNotification({
+            message,
+            type: NOTIFICATION_TYPES.ERROR,
+          }),
+        );
+      });
+  }, []);
+  useEffect(() => {
+    if (optionValue === CURRENT_EXECUTION_ONLY) {
+      setModalState({
+        testItems: [],
+        selectedItems: [],
+      });
+    } else {
+      const { searchMode } = modalState;
+      const requestData = {
+        searchMode,
+      };
+      if (searchMode === SEARCH_MODES.WITH_FILTER) {
+        requestData.filterId = activeFilter.id;
+      }
+      setLoading(true);
+      const itemIds = [];
+      let request;
+      if (optionValue === ALL_LOADED_TI_FROM_HISTORY_LINE) {
+        historyItems.forEach(({ id }) => itemIds.push(id));
+        request = fetch(URLS.bulkLastLogs(activeProject), {
+          method: 'post',
+          data: { itemIds, logLevel: 'ERROR' },
+        });
+      }
+      if (searchMode) {
+        request = fetch(URLS.logSearch(activeProject, currentTestItems[0].id), {
+          method: 'post',
+          data: requestData,
+        });
+      }
+      request
+        .then((response) => {
+          const similarItemsWithLogs = [];
+          if (optionValue === ALL_LOADED_TI_FROM_HISTORY_LINE) {
+            historyItems.forEach((item) =>
+              similarItemsWithLogs.push({
+                ...item,
+                itemId: item.id,
+                logs: response[item.id],
+                opened: SHOW_LOGS_BY_DEFAULT,
+              }),
+            );
+          } else {
+            similarItemsWithLogs.push(
+              ...response.map((item) => ({
+                ...item,
+                id: item.itemId,
+                opened: SHOW_LOGS_BY_DEFAULT,
+              })),
+            );
+          }
+          setModalState({
+            testItems: similarItemsWithLogs,
+            selectedItems: [],
+          });
+          setLoading(false);
+        })
+        .catch(({ message }) => {
+          setModalState({
+            testItems: [],
+            selectedItems: [],
+          });
+          setLoading(false);
+          dispatch(
+            showNotification({
+              message,
+              type: NOTIFICATION_TYPES.ERROR,
+            }),
+          );
+        });
+    }
+  }, [optionValue]);
+
+  return (
+    <>
+      <div className={cx('header')}>{formatMessage(messages.executionToChange)}</div>
+      {currentTestItems.map((item) => (
+        <Fragment key={item.id}>
+          <TestItemDetails
+            item={item}
+            logs={item.logs}
+            showErrorLogs={currentItemsLoading || item.opened}
+            loading={currentItemsLoading}
+          />
+        </Fragment>
```

Comment:
There is no need to use `Fragment` here, you can provide `key` prop direcrly to the component.